### PR TITLE
Add tests for models

### PR DESCRIPTION
**Pull request by WoosukKwon**
This PR adds basic tests for the vLLM models (that can run on a single GPU). The tests compare the outputs of HF and vLLM when using greedy sampling. NOTE: vLLM currently does not pass some of the tests.

@zhuohan123 I think you can extend `HfRunner` and `VllmRunner` for the tests for #857.

TODOs (in the future PRs):
- [ ] Make vLLM pass all the tests.
- [ ] Add tests for `tensor_parallel_size` > 1.
- [ ] Cache HF outputs to speed up the test execution.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    def generate_greedy(
+        self,
+        prompts: List[str],
+        max_tokens: int,
+    ) -> List[Tuple[List[int], str]]:
+        return self.generate(prompts, do_sample=False,
+                             max_new_tokens=max_tokens)
+
+
+@pytest.fixture
```
Comment (zhuohan123) [inline tests/conftest.py:84]: What is the main reason to add `@pytest.fixture` decorator compared to directly use these runners in the test function?

File: tests/models/test_models.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def test_falcon(hf_runner, vllm_runner, example_prompts) -> None:
+    _test_model(
+        hf_runner, vllm_runner, example_prompts, model="tiiuae/falcon-7b")
+
+
+def test_llama(hf_runner, vllm_runner, example_prompts) -> None:
+    _test_model(hf_runner, vllm_runner, example_prompts,
+                model="meta-llama/Llama-2-7b-hf")
```
Comment (zhuohan123) [inline tests/models/test_models.py:74]: Can you use [`@pytest.mark.parametrize`](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) to parametrize these tests?

**Review by zhuohan123 — CHANGES_REQUESTED** [review]
Thanks! Left some comments.

File: tests/conftest.py (hunk 1 above)
Comment (WoosukKwon) [reply tests/conftest.py:84]: To my understanding, there's no good way to import these runners into tests. Please let me know if there is.

File: tests/models/test_models.py (hunk 2 above)
Comment (WoosukKwon) [reply tests/models/test_models.py]: Good catch! I didn't consider it. Now I cleaned up the code to use `@pytest.mark.parametrize`.

**Comment by WoosukKwon** [comment]
@zhuohan123 Thanks for your comments. I've updated the PR. PTAL.

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thanks for the PR!

**Pull request merged**